db/models/aggregate_functions.py

```diff
@@ -153,9 +153,31 @@ def submit_application(application_id):
 def update_form(application_id, form_name, question_json):
     try:
         form_sql_row = FormsMethods.get_form(application_id, form_name)
+        # Running update form for the first time
+        if question_json and not form_sql_row.json:
+            ApplicationsMethods.application_edited(application_id)
+            current_app.logger.info(
+                f"Application updated for application_id: '{application_id}."
+            )
+        # Removing all data in the form
+        elif form_sql_row.json and not question_json:
+            ApplicationsMethods.application_edited(application_id)
+            current_app.logger.info(
+                f"Application updated for application_id: '{application_id}."
+            )
+        # Updating form subsequent times
+        elif (
+            form_sql_row.json
+            and form_sql_row.json[0]["fields"] != question_json[0]["fields"]
+        ):
+            ApplicationsMethods.application_edited(application_id)
+            current_app.logger.info(
+                f"Application updated for application_id: '{application_id}."
+            )
+    except sqlalchemy.orm.exc.NoResultFound as e:
+        raise e
+    finally:
         form_sql_row.json = question_json
-        db.session.commit()
         update_statuses(application_id, form_name)
+        db.session.commit()
```

Comment:
great spot Gio (brings us to the point of [ACID principles](https://www.oracle.com/in/database/what-is-a-relational-database/#:~:text=of%20the%20business.-,ACID%20properties%20and%20RDBMS,-Four%20crucial%20properties)). We should be transactional in our db queries (something we previously though flask-sql-alchemy handled automatically in request lifecycles but on initial testing this doesnt seem to be the case), a db.session.commit() appears to commit regardless of any subsequent failures in the flask request. @gidsg / @stevewalton28  I suspect this is one to watch/action as the flask apps dont seem to have transaction level management, this isnt causing issues currently AFAIK but could cause some if not better managed within flask apps. Possible solution maybe something like [this](https://docs.sqlalchemy.org/en/13/orm/session_basics.html#:~:text=%23%23%23%20another%20way%20(but%20again%20*not%20the%20only%20way*)%20to%20do%20it%20%23%23%23). 🎏